### PR TITLE
[SAIC-608] Exclude deleted tenant

### PR DESCRIPTION
**Pull request by Settis**
Exclude deleted tenant while check existing VMs on DST.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.filter_similar_vms_from_dst
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:45: [W1201(logging-not-lazy), FilterSimilarVMsFromDST.run] Specify string format arguments as logging function parameters
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:50: [W1201(logging-not-lazy), FilterSimilarVMsFromDST.run] Specify string format arguments as logging function parameters
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:63: [W0201(attribute-defined-outside-init), FilterSimilarVMsFromDST.find_similar_instances] Attribute 'dst_instances' defined outside __init__
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:39: [W0201(attribute-defined-outside-init), FilterSimilarVMsFromDST.run] Attribute 'tenant_id_to_new_id' defined outside __init__
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:65: [W0201(attribute-defined-outside-init), FilterSimilarVMsFromDST.find_similar_instances] Attribute 'similar_isntances' defined outside __init__
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:38: [W0201(attribute-defined-outside-init), FilterSimilarVMsFromDST.run] Attribute 'src_instances' defined outside __init__
cloudferrylib/os/actions/filter_similar_vms_from_dst.py:66: [W0201(attribute-defined-outside-init), FilterSimilarVMsFromDST.find_similar_instances] Attribute 'conflict_instances' defined outside __init__

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/filter_similar_vms_from_dst.py [hunk 1]
```diff
@@ -65,7 +65,9 @@ def find_similar_instances(self):
         self.similar_isntances = collections.defaultdict(set)
         self.conflict_instances = collections.defaultdict(set)
         for tenant_id, ip_to_id in titii_src.items():
-            tenant_new_id = self.tenant_id_to_new_id[tenant_id]
+            tenant_new_id = self.tenant_id_to_new_id.get(tenant_id, None)
+            if tenant_new_id is None:
+                continue
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/filter_similar_vms_from_dst.py]: This change introduces inconsistency: for most objects `[<cloud>] tenant` config value is used if object's tenant was deleted, while this patch simply ignores objects which belong to deleted tenant. I think correct resolution should be based either on properly maintaining contents of `src_cloud['identity_info']['tenants']['meta']['new_id']` values or reusing `cloudferrylib.os.identity.keystone.get_dst_user_from_src_user_id()` or something similar.

**Comment by roman-verchikov** [comment]
-1, please see comment and refactor accordingly

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by MirantisWorkloadMobility** [comment]
Can one of the admins verify this patch?

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**